pkg/publish/ingress.go

```diff
@@ -104,13 +104,19 @@ func toEndpoint(pattern, domain, container, appName, appNamespace string) (strin
 	return templateBuffer.String(), nil
 }
 
-func getAppHash(serviceName, appName, appNamespace string) string {
+/*
+getAppFullName takes containerName, appName and namespace, convert it to the format of
+`${containerName}-${appName}-${hash of `containerName:appName:namespace`}, with
+the ability to specify the length of the hash suffix
+*/
+func getAppFullName(serviceName, appName, appNamespace string, limit int) string {
```

Comment:
made it a go template function so that user can override the limit as they want.